### PR TITLE
Add `StorageBackend::size()` method

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-storage/bee-storage/src/backend.rs [hunk 1]
```diff
@@ -18,8 +16,11 @@ pub trait StorageBackend: Send + Sized + Sync + 'static {
...
 
     /// Initializes and starts the backend.
-    async fn start(config: Self::Config) -> Result<Self, Box<dyn Error>>;
+    async fn start(config: Self::Config) -> Result<Self, Self::Error>;
 
     /// Shutdowns the backend.
-    async fn shutdown(self) -> Result<(), Box<dyn Error>>;
+    async fn shutdown(self) -> Result<(), Self::Error>;
+
+    /// Size of the database in bytes.
```
Comment (zesterer) [inline bee-storage/bee-storage/src/backend.rs]: What does this mean?

I mean, I understand that it's talking about disk space, but is this the file the database is backed by? What should backends that don't support sizes (like a hypothetical `NullBackend`) return?

Comment (thibault-martinez) [reply bee-storage/bee-storage/src/backend.rs]: I'd say that `NullBackend::size()` should return 0!

**Review by louaykamel — DISMISSED** [review]

**Review by samuel-rufi — DISMISSED** [review]

File: bee-storage/bee-storage/src/backend.rs (hunk 1 above)
Comment (Alex6323) [reply bee-storage/bee-storage/src/backend.rs]: I feel like this PR could be motivated a little better. I want to ack this, but I have no clue about why and where it's necessary. Why wasn't it considered necessary before? What changed?

Comment (thibault-martinez) [reply bee-storage/bee-storage/src/backend.rs]: We just didn't think about it. Necessary for the dashboard that displays it.

Comment (Alex6323) [reply bee-storage/bee-storage/src/backend.rs]: :ok_hand:

**Review by zesterer — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**